gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -52,6 +54,21 @@
 
   public static final String HDFS_ILLEGAL_TOKEN_REGEX = "[\\s:\\\\]";
 
+  /**
+   * A {@link Collection} of all S3 schemes.
+   *
+   * <p>
+   *   The following important properties are useful to remember when writing code that is compatible with S3:
+   *   <ul>
+   *     <li>Renames are not atomic, and require copying the entire source file to the destination file</li>
+   *     <li>Writes to S3 will first go to the local filesystem, when the stream is closed the local file will be
```

Comment:
This is technically not true if you use s3a and call moveFromLocalFile.  In that case, the file is not written to a temp file locally and then copied up.